### PR TITLE
feat: initial document audit logs implementation

### PR DESCRIPTION
**Pull request by dguyen**
Added the initial implementation of document audit logs.

There are currently no visual UI changes for this. This is purely to begin logging any logs related to document changes and events which will allow us to iterate and test things.

### CODE REVIEW

File: packages/prisma/schema.prisma [hunk 1]
```diff
@@ -170,11 +170,30 @@ model Document {
...
 
+  auditLogs DocumentAuditLog[]
+
   @@unique([documentDataId])
   @@index([userId])
   @@index([status])
 }
 
+model DocumentAuditLog {
+  id         Int      @id @default(autoincrement())
```
Comment (Mythie) [inline packages/prisma/schema.prisma:181]: use cuid pls

File: packages/lib/types/document-audit-logs.ts [hunk 2]
```diff
@@ -0,0 +1,346 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// Be aware that any changes to this file may need migrations since we're store JSON in Prisma.
```
Comment (Mythie) [inline packages/lib/types/document-audit-logs.ts:3]: Woolies or coles?

File: packages/lib/types/document-audit-logs.ts [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+  'RECIPIENT_CREATED',
+  'RECIPIENT_DELETED',
+  'RECIPIENT_UPDATED',
+
+  // Document events.
+  'DOCUMENT_COMPLETED',
+  'DOCUMENT_CREATED',
+  'DOCUMENT_DELETED',
+  'DOCUMENT_FIELD_SIGNED',
+  'DOCUMENT_FIELD_UNSIGNED',
```
Comment (Mythie) [inline packages/lib/types/document-audit-logs.ts:27]: DOCUMENT_FIELD_INSERTED
DOCUMENT_FIELD_REMOVED

File: packages/lib/types/document-audit-logs.ts [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+  'DOCUMENT_META_UPDATED',
+  'DOCUMENT_OPENED',
+  'DOCUMENT_TITLE_UPDATED',
+  'DOCUMENT_RECIPIENT_FLOW_COMPLETE',
```
Comment (Mythie) [inline packages/lib/types/document-audit-logs.ts:31]: DOCUMENT_RECIPIENT_COMPLETED?
RECIPIENT_COMPLETED?
RECIPIENT_SIGNING_COMPLETED?

Jus ideas

Comment (Mythie) [reply packages/lib/types/document-audit-logs.ts:31]: DOCUMENT_RECIPIENT_SUBMITTED

File: packages/prisma/migrations/20240209023519_add_document_audit_logs/migration.sql [hunk 5]
```diff
@@ -6,7 +6,13 @@
...
 */
 -- AlterTable
-ALTER TABLE "Field" ADD COLUMN     "secondaryId" TEXT NOT NULL;
+ALTER TABLE "Field" ADD COLUMN     "secondaryId" TEXT;
+
+-- Set all null secondaryId fields to a uuid
+UPDATE "Field" SET "secondaryId" = gen_random_uuid()::text WHERE "secondaryId" IS NULL;
+
+-- Restrict the Field to required
+ALTER TABLE "Field" ALTER COLUMN "secondaryId" SET NOT NULL;
```
Comment (dguyen) [inline packages/prisma/migrations/20240209023519_add_document_audit_logs/migration.sql:15]: @Mythie could you please have a look at this custom migration change to backfill `secondaryId`

We can't use cuid in migrations, so I'm using UUID

This means that we're going to have two different types of IDs for `secondaryId`, is that fine?

Comment (Mythie) [reply packages/prisma/migrations/20240209023519_add_document_audit_logs/migration.sql:15]: That's fine, lgtm

**Pull request merged**